src/compiler/checker.ts

```diff
@@ -14749,19 +14749,38 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
      * maps primitive types and type parameters are to their apparent types.
      */
     function getSignaturesOfType(type: Type, kind: SignatureKind): readonly Signature[] {
-        const result = getSignaturesOfStructuredType(getReducedApparentType(type), kind);
-        if (kind === SignatureKind.Call && !length(result) && type.flags & TypeFlags.Union) {
+        function carveoutResult(type: Type, kind: SignatureKind): readonly Signature[] | undefined {

```

Comment:
The carve out for arrays is raised in precedence.  Previously it only occurred after `result = getSignaturesOfStructuredType` was called, and only if `length(result)` was zero.
It squeezes the (# Overloads X # Template Params) grid of signatures to a (# Overloads X 1) set of overloads by changing, e.g. an (A[] | B[]) union of arrays to a non-union ((A|B)[])  array of a single type union.  Thus there is in increase in such changes in the tests results.